unified_planning/model/timing.py

```diff
@@ -212,6 +218,20 @@ def __init__(self, lower: FNode, upper: FNode, is_left_open: bool = False, is_ri
         Duration.__init__(self)
         Interval.__init__(self, lower, upper, is_left_open, is_right_open)
 
+    def __eq__(self, oth: object) -> bool:
+        if isinstance(oth, DurationInterval):
+            return self._lower == oth._lower and self._upper == oth._upper and self._is_left_open == oth._is_left_open and self._is_right_open == oth._is_right_open
+        else:
+            return False
+
+    def __hash__(self) -> int:
+        res = hash(self._lower) + hash(self._upper)
+        if self._is_left_open:
+            res ^= hash('is_left_open')
```

Comment:
OK, but why using a string here, you can easily hash booleans as well (booleans are numeric types in python)